rand/id.go

```diff
@@ -0,0 +1,68 @@
+package rand
+
+import (
+	"encoding/binary"
+	"math/rand"
+	"sync"
+
+	"github.com/influxdata/influxdb"
+)
+
+var _ influxdb.IDGenerator = (*OrgBucketID)(nil)
+
+// OrgBucketID creates an id that does not have ascii
+// backslash, commas, or spaces.  Used to create IDs for organizations
+// and buckets.
+//
+// It is implemented without those characters because orgbucket
+// pairs are placed in the old measurement field.  Measurement
+// was interpreted as a string delimited with commas.  Therefore,
+// to continue to use the underlying storage engine we need to
+// sanitize ids.
+//
+// Safe for concurrent use by multiple goroutines.
+type OrgBucketID struct {
+	m   sync.Mutex
+	src *rand.Rand
+}
+
+// NewOrgBucketID creates an influxdb.IDGenerator that creates
+// random numbers seeded with seed.  Ascii backslash, comma,
+// and space are manipulated by incrementing.
+//
+// Typically, seed with `time.Now().UnixNano()`
+func NewOrgBucketID(seed int64) *OrgBucketID {
+	return &OrgBucketID{
+		src: rand.New(rand.NewSource(seed)),
+	}
+}
+
+// Seed allows one to override the current seed.
+// Typically, this override is done for tests.
+func (r *OrgBucketID) Seed(seed int64) {
+	r.m.Lock()
+	r.src = rand.New(rand.NewSource(seed))
+	r.m.Unlock()
+}
+
+// ID generates an ID that does not have backslashes, commas, or spaces.
+func (r *OrgBucketID) ID() influxdb.ID {
+	r.m.Lock()
```

Comment:
If this ever becomes too slow, we can pre-buffer n random ids that only need read access to distribute.